File: app/src/main/java/org/mifos/mobilewallet/data/fineract/entity/mapper/ClientDetailsMapper.java
package org.mifos.mobilewallet.data.fineract.entity.mapper;

import org.mifos.mobilewallet.data.fineract.entity.client.Client;
import org.mifos.mobilewallet.home.domain.model.ClientDetails;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by naman on 10/7/17.
 */

@Singleton
public class ClientDetailsMapper {

    @Inject
    public ClientDetailsMapper() {}

    public ClientDetails transform(Client client) {
        ClientDetails clientDetails = new ClientDetails();

        if (client != null) {
            clientDetails.setName(client.getDisplayName());
            clientDetails.setClientId(client.getId());
            clientDetails.setExternalId(client.getExternalId());

        }
        return clientDetails;
    }
}

File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/ClientDetailsMapper.kt
package org.mifos.mobilewallet.core.data.fineract.entity.mapper

import com.mifos.mobilewallet.model.entity.client.Client 
import javax.inject.Inject
import com.mifos.mobilewallet.model.domain.client.Client as DomainClient
/**
 * Created by naman on 10/7/17.
 */
class ClientDetailsMapper @Inject constructor() {
    fun transformList(clients: List<Client?>?): List<DomainClient> {
        val clientList: MutableList<DomainClient> = ArrayList()
        clients?.forEach { client ->
            clientList.add(transform(client))
        }
        return clientList
    }

    fun transform(client: Client?): DomainClient {
        val clientDetails = DomainClient()
        if (client != null) {
            clientDetails.name = client.displayName
            clientDetails.clientId = client.id.toLong()
            clientDetails.externalId = client.externalId
            clientDetails.mobileNo = client.mobileNo
        }
        return clientDetails
    }
}
